project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectServiceImpl.java

```diff
@@ -331,6 +345,57 @@ public Mono<Sample> findSample(String projectId, String sampleId) {
     }).subscribeOn(scheduler);
   }
 
+  @Override
+  public Flux<OntologyTerm> getTerms(String value, int offset, int limit) {
+    return Flux.defer(() -> {
+      try {
+        return Flux.fromIterable(terminologyService.search(value, offset, limit))
+            .map(AsyncProjectServiceImpl::convertToApi);
+      } catch (Exception e) {
+        log.error("Error searching for term " + value, e);
+        return Flux.error(new RequestFailedException("Error searching for term " + value));
+      }
+    }).subscribeOn(scheduler);
+  }
+
+  @Override
+  public Mono<OntologyTerm> getTermWithCurie(Curie value) {
+    return Mono.defer(() -> {
+      try {
+        return Mono.justOrEmpty(terminologyService.findByCurie(value.toString()))
+            .map(AsyncProjectServiceImpl::convertToApi);
+      } catch (Exception e) {
+        log.error("Error searching for term " + value, e);
+        return Mono.error(new RequestFailedException("Error searching for term " + value));
+      }
```

Comment:
The same considerations of making the Mono creation simpler and moving the error handling to the framework methods.